### PR TITLE
Add react-paginate module and logic for pagination

### PR DESCRIPTION
**Pull request by dorkka**

### CODE REVIEW

File: src/components/ResortsData.js [hunk 1]
```diff
@@ -18,30 +20,37 @@ class ResortsData extends Component{
...
+    //there is +1, because 'react-paginate' has 0 as start page number,
+    this.setState({page: data.selected +1})
+  };
 
   render(){
-    const {isLoading, data, error} = this.state;
+    const {data, pageCount, isLoading, error} = this.state;
     if(error)
       return(error.message)
     if(isLoading)
```
Comment (ar-mac) [inline src/components/ResortsData.js]: When you click pagination it triggers loading flag, and the whole main return (which contains pagination) is removed and replaced with `Loading in progress`.
When data is loaded, main return (and Pagination component) is rendered again, and by default the page number `1` is selected. That's why it was always displayed regardless of what you clicked.

BTW If you click twice on the same page, the active button properly changes.

Comment (dorkka) [reply src/components/ResortsData.js]: thx!
yes I knew that when I click twice 'active' is triggered :)

File: src/components/ResortsData.js [hunk 2]
```diff
@@ -18,30 +20,34 @@ class ResortsData extends Component{
...
           if (response.ok) {
-            return response.json();
+            return response.json().then((data) => ({data, headers: response.headers}))
           } else {
             throw new Error('Something went wrong ...');
           }
         })
-        .then(data => this.setState({ data: data, isLoading: false }))
+        .then(({data, headers}) => {
+          this.setState({ data: data,  pageCount: Math.ceil(headers.get('x-total-count')/ this.state.perPage), isLoading: false })
```
Comment (ar-mac) [inline src/components/ResortsData.js]: (minor)
This line is quite long. Please split params to separate line.
Also, use shorthand for data.

**Review by ar-mac — APPROVED** [review]

File: src/components/ResortsData.js [hunk 3]
```diff
@@ -37,7 +37,8 @@ class ResortsData extends Component{
           }
         })
         .then(({data, headers}) => {
-          this.setState({ data: data,  pageCount: Math.ceil(headers.get('x-total-count')/ this.state.perPage), isLoading: false })
+          this.setState(
+           {data, pageCount: Math.ceil(headers.get('x-total-count')/this.state.perPage), isLoading: false})
```
Comment (ar-mac) [inline src/components/ResortsData.js]: what I meant, in such cases to shorten the lines split them like that.

```
this.setState({
  data,
  pageCount: Math.ceil(headers.get('x-total-count')/this.state.perPage),
  isLoading: true,
})
```

Comment (dorkka) [reply src/components/ResortsData.js]: OK :)

**Pull request merged**